hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstant.java

```diff
@@ -30,7 +34,23 @@
  *
  * @see HoodieTimeline
  */
-public class HoodieInstant implements Serializable {
+public class HoodieInstant implements Serializable, Comparable<HoodieInstant> {
+
+  /**
+   * A COMPACTION action eventually becomes COMMIT when completed. So, when grouping instants
+   * for state transitions, this needs to be taken into account
+   */
+  private static final Set<String> comparableActions = Arrays.stream(new String[] { HoodieTimeline.COMMIT_ACTION,
+      HoodieTimeline.COMPACTION_ACTION}).collect(Collectors.toSet());
+
+  public static final Comparator<HoodieInstant> comparator = Comparator.comparing(HoodieInstant::getTimestamp)
```

Comment:
How does this ensure choosing the highest state ? Is it just string comparison ?

Reply:
+1 on this